lib/base-compiler.js

```diff
@@ -258,6 +262,65 @@ class BaseCompiler {
         return tooling;
     }
 
+    buildExecutable(compiler, options, inputFilename, execOptions) {
+        // default implementation, but should be overridden by compilers
+        return this.runCompiler(compiler, options, inputFilename, execOptions);
+    }
+
+    getOrBuildExecutable(key) {
+        // todo: ask cache for executable if it already exists
```

Comment:
:+1: here . This is the main reason the S3 cache exists: then a compile on one node can be executed on another. And eventually we can separate execution onto different nodes entirely (the `sandbox` call might eventually send a message to a separate node with the hash as a parameter, for example).

It shouldn't be too tricky to check the cache here I _think_, right? But of course we need to be careful to use the right things in the key, and we need to ensure the resulting cache answer is executable...so I see why you might punt on this for now :)